src/test/java/com/synopsys/integration/jenkins/JenkinsProxyHelperTest.java

```diff
@@ -0,0 +1,94 @@
+    final int proxyPort = 13;
+
+    private static Stream<Arguments> populateNoProxyInfoTests() {
+        String hostToExclude = "www.exclude.com";
+        Pattern hostExcludePattern = Pattern.compile(hostToExclude);
+        List<Pattern> ignoredProxyHosts = new ArrayList<>();
+        ignoredProxyHosts.add(hostExcludePattern);
+
+        return Stream.of(
+            Arguments.of("", ignoredProxyHosts),
+            Arguments.of(null, ignoredProxyHosts),
+            Arguments.of("https://" + hostToExclude, ignoredProxyHosts)
+        );
+    }
+
+    private static Stream<Arguments> populateValidProxyInfoTests() {
+        String hostToInclude = "https://www.include.com";
+        String hostToExclude = "www.exclude.com";
+        Pattern hostExcludePattern = Pattern.compile(hostToExclude);
+
+        return Stream.of(
+            Arguments.of(hostToInclude, new ArrayList<>()),
+            Arguments.of(hostToInclude, null),
+            Arguments.of(hostToInclude, Arrays.asList(hostExcludePattern))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("populateNoProxyInfoTests")
+    public void testGetProxyInfo_NoProxyInfo(String url, List<Pattern> ignoredProxyHosts) {
+        // The returned proxyInfo object should be equal to ProxyInfo.NO_PROXY_INFO
+        ProxyInfo proxyInfo = JenkinsProxyHelper.getProxyInfo(url, proxyHost, proxyPort, proxyUsername, proxyPassword, ignoredProxyHosts, ntlmDomain, ntlmWorkstation);
+        assertEquals(ProxyInfo.NO_PROXY_INFO.getProxy().isPresent(), proxyInfo.getProxy().isPresent());
+        assertEquals(ProxyInfo.NO_PROXY_INFO.getHost().isPresent(), proxyInfo.getHost().isPresent());
+        assertEquals(ProxyInfo.NO_PROXY_INFO.getPort(), proxyInfo.getPort());
+        assertEquals(ProxyInfo.NO_PROXY_INFO.getProxyCredentials().isPresent(), proxyInfo.getProxyCredentials().isPresent());
+        assertEquals(ProxyInfo.NO_PROXY_INFO.getUsername(), proxyInfo.getUsername());
+        assertEquals(ProxyInfo.NO_PROXY_INFO.getPassword(), proxyInfo.getPassword());
+        assertEquals(ProxyInfo.NO_PROXY_INFO.getNtlmDomain(), proxyInfo.getNtlmDomain());
+        assertEquals(ProxyInfo.NO_PROXY_INFO.getNtlmWorkstation(), proxyInfo.getNtlmWorkstation());
+    }
+
+    @ParameterizedTest
+    @MethodSource("populateValidProxyInfoTests")
+    public void testGetProxyInfo_PopulatedProxyInfo(String url, List<Pattern> ignoredProxyHosts) {
+        // The returned proxyInfo object should be populated with the values supplied to getProxyInfo
+        ProxyInfo proxyInfo = JenkinsProxyHelper.getProxyInfo(url, proxyHost, proxyPort, proxyUsername, proxyPassword, ignoredProxyHosts, ntlmDomain, ntlmWorkstation);
+        assertTrue(proxyInfo.getProxy().isPresent());
+        assertTrue(proxyInfo.getHost().isPresent());
+        assertEquals(proxyHost, proxyInfo.getHost().orElse(null));
+        assertEquals(proxyPort, proxyInfo.getPort());
+        assertTrue(proxyInfo.getProxyCredentials().isPresent());
```

Comment:
On lines 75-79, you are checking that the optionals contain values and unwrapping the values to check their contents. On lines 80-84, you are comparing optionals directly. Both are valid approaches, but combining them is sure to be confusing to the next developer-- I suggest you pick one and stick to it!